File: sample-iOS/Supr-demo/Supr-demo/Supr/Domain/UseCase.swift
//
//  UseCase.swift
//  Supr-demo
//
//  Created by Arnaud Camus on 2/22/18.
//  Copyright © 2018 Impraise. All rights reserved.
//

import RxSwift

protocol UseCase {
    var subscriptions: CompositeDisposable { get set }
}

File: sample-Android/Suprdemo/supr/src/main/java/com/impraise/supr/domain/UseCase.kt
package com.impraise.supr.domain

import io.reactivex.disposables.CompositeDisposable

/**
 * Created by guilhermebranco on 1/16/18.
 */
abstract class UseCase<Object>: DisposableUseCase {

    protected val subscriptions = CompositeDisposable()

    var callback: ((result: Object) -> Unit)? = null

    override fun dispose() {
        if (!subscriptions.isDisposed) {
            subscriptions.dispose()
            subscriptions.clear()
        }
    }
}